src/foam/dao/OnDeleteRemoveChildrenDAO.js

```diff
@@ -0,0 +1,58 @@
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'OnDeleteRemoveChildrenDAO',
```

Comment:
How about 'RemoveChildrenOnRemoveDAO' instead. Since the DAO method is called 'remove' not 'delete'.